File: Menus/PauseMenu.js
// This Script control the Pause Screen .
var pauseUI : GameObject;
var paused : boolean;
var gamePad = false;

function Update() {
    if(Input.GetJoystickNames().Length > 0) {
        if(Input.GetJoystickNames()[0].Length > 0) {
            gamePad = true;
        }
        else {
            gamePad = false;
        }
    }

    pauseUI.GetComponent(AudioSource).ignoreListenerPause = true;
    if (Input.GetButtonDown("Start")) {
        paused = !paused;
    }
    if (paused) {
        Pause();
    }
    else {
        Resume();
    }
}

function Pause() {
    Time.timeScale = 0;
    AudioListener.pause = true;
    pauseUI.SetActive(true);
    if(GameObject.Find("Player") != null) {
        GameObject.Find("Player").GetComponent(PlayerControl).enabled = false;
    }
    if(GameObject.Find("Player Intro") != null) {
        GameObject.Find("Player Intro").GetComponent(PlayerControlIntro).enabled = false;
    }
    if (Input.GetButtonDown("Select")) {
        Application.Quit();
    }
    if (gamePad) {
        GameObject.Find("JoyCon Text").GetComponent(Image).enabled = true;
        GameObject.Find("KeyCon Text").GetComponent(Image).enabled = false;
    }
    else {
        GameObject.Find("JoyCon Text").GetComponent(Image).enabled = false;
        GameObject.Find("KeyCon Text").GetComponent(Image).enabled = true;
    }
}

function Resume() {
    Time.timeScale = 1; 
    AudioListener.pause = false;
    pauseUI.SetActive(false);
    if(GameObject.Find("Player") != null) {
        GameObject.Find("Player").GetComponent(PlayerControl).enabled = true;
    }
    if(GameObject.Find("Player Intro") != null) {
        GameObject.Find("Player Intro").GetComponent(PlayerControlIntro).enabled = true;
    }
}

File: Menus/PauseMenu.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.UI;

public class PauseMenu : MonoBehaviour {
    public bool paused = false;
    Image joyCon;
    Image keyCon;
    GameObject pauseUI;
    GameObject player;
    PlayerControl playerControl;
    PlayerAnimation playerAnimation;
    PlayerAttack playerAttack;

    void Start() {
        player = GameObject.FindWithTag("Player");
        playerControl = player.GetComponent<PlayerControl>();
        playerAnimation = player.GetComponent<PlayerAnimation>();
        playerAttack = player.GetComponent<PlayerAttack>();
        pauseUI = GameObject.Find("PauseUI");
        pauseUI.GetComponent<AudioSource>().ignoreListenerPause = true;
        pauseUI.GetComponent<Canvas>().enabled = true;
        joyCon = GameObject.Find("JoyCon Text").GetComponent<Image>();
        keyCon = GameObject.Find("KeyCon Text").GetComponent<Image>();
        joyCon.enabled = false;
        keyCon.enabled = true;
    }

    void Update() {
        GamepadDetection();
        PauseDetection();
    }

    void GamepadDetection() {
        foreach (string gamePad in Input.GetJoystickNames()) {
            joyCon.enabled = gamePad == "" ? false : true;
            keyCon.enabled = gamePad == "" ? true : false;
        }
    }

    void PauseDetection() {
        if (Input.GetButtonDown("Start")) { paused = !paused; }
        if (paused) { Pause(); } else { Resume(); }
    }

    void Pause() {
        Time.timeScale = 0;
        AudioListener.pause = true;
        pauseUI.SetActive(true);
        playerControl.enabled = false;
        playerAnimation.enabled = false;
        playerAttack.enabled = false;
        if (Input.GetButtonDown("Select")) { Application.Quit(); }
    }

    void Resume() {
        Time.timeScale = 1;
        AudioListener.pause = false;
        pauseUI.SetActive(false);
        playerControl.enabled = true;
        playerAnimation.enabled = true;
        playerAttack.enabled = true;
    }
}